packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -0,0 +1,167 @@
+  )?.includes?.('device-auth');
+
+  return (
+    <>
+      <FormFieldContainer>
+        <TextInput
+          data-testid="connection-oidc-username-input"
+          onChange={({
+            target: { value },
+          }: React.ChangeEvent<HTMLInputElement>) => {
+            updateConnectionFormField({
+              type: 'update-username',
+              username: value,
+            });
+          }}
+          label="Principal"
+          optional
+          value={username || ''}
+          errorMessage={usernameError}
+          state={usernameError ? 'error' : undefined}
+        />
+      </FormFieldContainer>
+      <FormFieldContainer>
+        <Accordion text="OIDC Options">
+          <FormFieldContainer>
+            <TextInput
+              data-testid="connection-oidc-auth-code-flow-redirect-uri-input"
+              onChange={({
+                target: { value },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                handleFieldChanged('redirectURI', value);
+              }}
+              optional
+              label="Auth Code Flow Redirect URI"
+              value={connectionOptions?.oidc?.redirectURI || ''}
+            />
+          </FormFieldContainer>
+
+          <FormFieldContainer>
+            <Checkbox
+              onChange={({
+                target: { checked },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                if (checked) {
+                  updateConnectionFormField({
+                    type: 'update-auth-mechanism-property',
+                    key: 'ALLOWED_HOSTS',
+                    value: '*',
+                  });
+                  return;
+                }
+
+                // Unset the allowed hosts.
+                updateConnectionFormField({
+                  type: 'update-auth-mechanism-property',
+                  key: 'ALLOWED_HOSTS',
+                  value: '',
+                });
+              }}
+              data-testid="oidc-allow-untrusted-endpoint-input"
+              id="oidc-allow-untrusted-endpoint-input"
+              label={
+                <>
+                  <Label htmlFor="oidc-allow-untrusted-endpoint-input">
+                    Enable untrusted target endpoint
+                  </Label>
+                  <Description>
+                    Allow connecting when the target endpoint is not in the list
+                    of trusted endpoints &#40;this sets the driver&apos;s
+                    ALLOWED_HOSTS list to *&#41;
+                  </Description>
+                </>
+              }
+              checked={hasEnabledUntrustedEndpoints}
+            />
+          </FormFieldContainer>
+
+          <FormFieldContainer>
+            <Checkbox
+              onChange={({
+                target: { checked },
+              }: React.ChangeEvent<HTMLInputElement>) => {
+                if (checked) {
+                  handleFieldChanged('allowedFlows', ['device-auth']);
```

Comment:
Doesn’t this override `allowedFlows = ['auth-code']` to `allowedFlows = ['device-auth']` when checked, and then `allowedFlows = ['device-auth']` to `allowedFlows = []` when unchecked?

Reply:
Clarifying, from my understanding we'll need to supply a `notifyDeviceFlow` handler and show a ui for the device `userCode` in this work? https://github.com/mongodb-js/mongosh/pull/1458/files#diff-d998d2f6015503912abc1073f3bdb46da4c12651a7ba3d2e6170711edb19b45aR808
Since [`allowedFlows` is defaulted to include `auth-code` and `device-auth`](https://github.com/mongodb-js/oidc-plugin/blob/7bbbd1246833871491b75266193cd5a87bf845d5/src/plugin.ts#L266), shouldn't this be setting allowedFlows to only be `['device-auth']` to avoid `auth-code`?

Reply:
> I added one, can remove if it's not needed here.

No, it looks great! This is exactly what we want :+1:

> Since [`allowedFlows` is defaulted to include `auth-code` and `device-auth`](https://github.com/mongodb-js/oidc-plugin/blob/7bbbd1246833871491b75266193cd5a87bf845d5/src/plugin.ts#L266), shouldn't this be setting allowedFlows to only be `['device-auth']` to avoid `auth-code`?

I think we’ll want to default to `['auth-code']` in Compass like we do in mongosh, and I’d lean towards this button making it so that the value is `['auth-code', 'device-auth']` rather than `['device-auth']` only. We can probably re-evaluate that choice if it turns out to be an issue, but in practice it really shouldn’t be.

The default in the OIDC plugin is mostly that way because we made the decision to default to Auth-Code-Flow-only in mongosh fairly late (and it’s not currently a real security problem because we only connect to trusted endpoints anyway). Maybe also relevant: https://github.com/mongodb/docs-node/pull/698/files#r1210137439

Reply:
Updated to default 👍 
I'm thinking we'll want an e2e test to cover device auth in Compass, separate pr?